lib/AST/Module.cpp

```diff
@@ -1247,6 +1247,44 @@ bool ModuleDecl::isSystemModule() const {
   return false;
 }
 
+namespace {
+enum class SerializedModuleKind {
+  Other,
+  Clang,
+  Swift,
+};
+}
+static SerializedModuleKind getSerializedModuleKind(const ModuleDecl *M) {
+  Optional<SerializedModuleKind> K;
+
+  for (auto File : M->getFiles()) {
+    switch (File->getKind()) {
+      case FileUnitKind::Source:
+      case FileUnitKind::Builtin:
+      case FileUnitKind::Derived:
+        return SerializedModuleKind::Other;
+      case FileUnitKind::SerializedAST:
+        assert(!K.hasValue() && "cannot handle multi-file modules");
```

Comment:
Are these creating a ModuleDecl object containing mixed FileUnits and/or multiple serialized FileUnits ?
If yes, I could rename the functions to `isSingleClangModule`/`isSingleSwiftASTModule`
A module you will import in source code will be a single serialized AST unit, right ?